src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs

```diff
@@ -364,10 +364,782 @@ private void DisposeInternal()
+                        else
+                        {
+                            IntPtr[] intermediate = new IntPtr[namespaces.Length];
+                            fixed (IntPtr* intermediatePtr = intermediate)
+                            {
+                                hr = func(Inst, bufferLength, countPtr, intermediatePtr);
+                            }
+
+                            if (hr == 0)
+                            {
+                                for (int i = 0; i < count; i++)
+                                {
+                                    namespaces[i] = (SymUnmanagedNamespaceWrapperCache.SymUnmanagedNamespaceRcw)SymUnmanagedNamespaceWrapperCache.Instance.GetOrCreateObjectForComInstance(intermediate[i], CreateObjectFlags.UniqueInstance);
+                                }
+                            }
+                        }
+                    }
+
+                    return hr;
+                }
+
+                public int GetVariables(int bufferLength, out int count, ISymUnmanagedVariable[] variables)
+                {
+                    var func = (delegate* unmanaged<IntPtr, int, int*, IntPtr*, int>)(*(*(void***)Inst + 5));
+                    int hr;
+
+                    fixed (int* countPtr = &count)
+                    {
+                        if (variables == null)
+                        {
+                            hr = func(Inst, bufferLength, countPtr, null);
+                        }
+                        else
+                        {
+                            IntPtr[] intermediate = new IntPtr[variables.Length];
+                            fixed (IntPtr* intermediatePtr = intermediate)
+                            {
+                                hr = func(Inst, bufferLength, countPtr, intermediatePtr);
+                            }
+
+                            if (hr == 0)
+                            {
+                                for (int i = 0; i < count; i++)
+                                {
+                                    variables[i] = (SymUnmanagedVariableWrapperCache.SymUnmanagedVariableRcw)SymUnmanagedVariableWrapperCache.Instance.GetOrCreateObjectForComInstance(intermediate[i], CreateObjectFlags.UniqueInstance);
+                                }
+                            }
+                        }
+                    }
+
+                    return hr;
+                }
+
+                public void Dispose()
+                {
+                    DisposeInternal();
+                    GC.SuppressFinalize(this);
+                }
+
+                private void DisposeInternal()
+                {
+                    if (_disposed)
+                    {
+                        return;
+                    }
+                    _disposed = true;
+                    Marshal.Release(Inst);
+                }
+
+                ~SymUnmanagedNamespaceRcw()
+                {
+                    DisposeInternal();
```

Comment:
Didn't find out a proper way to dispose as need, so I leave the clean-up work to GC.